lib/pure/random.nim

```diff
@@ -635,25 +639,46 @@ proc shuffle*[T](x: var openArray[T]) =
 
 when not defined(nimscript) and not defined(standalone):
   import times
+  
+  proc initRand(): Rand =
+    ## Initializes a new Rand state with a seed based on the current time.
+    ##
+    ## The resulting state is independent of the default random number generator's state.
+    ##
+    ## **Note:** Does not work for NimScript or the compile-time VM.
```

Comment:
how about instead honor experimental:vmops:
if `experimental:vmops` is passed, it would use it, else would use a fixed seed (ie reproducible builds)

Reply:
Well you wouldn't use the `random` module if you cared about predictability. Current compilation isn't pure either, `IOEffect` procs already exist in VM (though not tagged as such), this would just use a `TimeEffect` proc. There's no way to check for that experimental, `compileOption` doesn't work for experimentals, `compiles(getTime())` is true under `when nimvm` even if it doesn't work. Maybe it should be a compile option like `--pureVM` or `--impureVM`, but even then the entire `random` module wouldn't make sense as being pure, other languages have things like `RandomEffect` (Idris and I believe some Haskell libraries).

Reply:
> There's no way to check for that experimental

refs https://github.com/nim-lang/Nim/issues/8644 which is very fixable

Reply:
anyways, this can be fixed in future PR